plugins/browser-plugin-link-click-tracking/test/events.test.ts

```diff
@@ -36,17 +36,15 @@ const getUEEvents = F.compose(F.filter(F.compose(F.eq('ue'), F.get('evt.e'))));
 const extractEventProperties = F.map(F.compose(F.get('data'), (cx: string) => JSON.parse(cx), F.get('evt.ue_pr')));
 const extractUeEvent = (schema: string) => {
   return {
-    from: F.compose(
-      F.first,
-      F.filter(F.compose(F.eq(schema), F.get('schema'))),
-      F.flatten,
-      extractEventProperties,
-      getUEEvents
-    ),
+    from: (a: any, n: number = 0) =>
+      F.nth(
+        n,
+        F.compose(F.filter(F.compose(F.eq(schema), F.get('schema'))), F.flatten, extractEventProperties, getUEEvents)(a)
+      ),
   };
 };
 
-describe('AdTrackingPlugin', () => {
+describe('LinkClickTrackingPlugin', () => {
```

Comment:
The link click tracking plugin has one of the lowest unit test coverage in the tracker (around 18% last time we checked it). This is a bit increasing the scope of the PR, but would you consider adding a few unit tests to cover more of the event listening? Considering how flaky the integration tests are, we wanted to move more of the testing to the unit tests. Peter has done some unit tests in the cross domain plugin that use jest to mock links and also test link clicks ([see here](https://github.com/snowplow/snowplow-javascript-tracker/blob/master/libraries/browser-tracker-core/test/tracker/cross_domain.test.ts#L53)). Maybe we could do something similar here if you have the capacity left?

Reply:
Very nice tests, thanks! I think @igneel64 used [istanbul.js](https://istanbul.js.org) to calculate the code coverage, there is [a report here](https://www.notion.so/keep-in-the-snow/Trackers-Integration-Testing-overview-Web-focused-cae853ad6ff2452da7275479ec99976f).